kotlinx-coroutines-core/common/src/flow/operators/Context.kt

```diff
@@ -150,6 +150,39 @@ public fun <T> Flow<T>.buffer(capacity: Int = BUFFERED): Flow<T> {
 @ExperimentalCoroutinesApi
 public fun <T> Flow<T>.conflate(): Flow<T> = buffer(CONFLATED)
 
+/**
+ * Discards flow emissions using a rendezvous channel. When applied, a producer will never suspend
+ * due to a slow collector. In the event that an element is emitted while a collector is busy, the
+ * emitted element is dropped.
+ *
+ * For example, consider a flow that emits integers 1 to 30 with a 100ms delay.
+ *
+ * ```
+ * val flow = (1..30).asFlow().onEach { delay(100) }
+ * ```
+ *
+ * Applying the `dropWhileBudy` operator allows a collector which takes 1 second to process each
+ * element to reieve integers [1, 11, 21]
+ *
+ * ```
+ * val result = flow().dropWhileBusy().onEach { delay(1000) }.tolist()
+ * assertEquals(listOf(1, 11, 21), result)
+ * ```
+ *
+ * Not that the `dropWhileBusy` operator is extremely similar to [conflate]. However, the difference
+ * is that [conflate] always emits the latest elements, whereas `dropWhileBusy` doesn't. `dropWhileBusy`
+ * explicity only emits elements given that a collector is currently suspended and able to accept
+ * another element.
+ *
+ * ### Operator Fusion
+ *
+ * Adjacent applications of `dropWhileBusy`, [conflate]/[buffer], [channelFlow], [flowOn], [produceIn], and [broadcastIn] are
+ * always fused so that only one properly configured channel is used for execution.
+ * **Conflation takes precedence over `buffer()` calls with any other capacity.**
+ */
+@ExperimentalCoroutinesApi
+public fun <T> Flow<T>.dropWhileBusy(): Flow<T> = DiscardFlowOperator(this)
```

Comment:
Follow up to this: 

What does it mean for us to fuse this operator with others? It seems like we could fuse this with `buffer`. Then, when the buffer of this channel gets full we would drop elements after. 

What would it mean for this to get fused with conflate? Seems like it shouldn't be a valid operation. Thoughts?

Reply:
Follow, follow up. 

I'm implemented operator fusion with `buffer`. It seems to make logical sense. 

I throw an `IllegalArgumentException` is `conflate` is called downstream of `dropWhileBusy` because it doesn't seem like a valid operation. This has been documented.

Reply:
The minimal strategy seems simple and to make sense. I do have concerns about the full strategy though. 

> It would be more efficient to directly offer into the rendezvous channel and use only two coroutines. That is, it would be more efficient if dropWhileBusy could fuse with the preceding ChannelFlowBuilder. But that means that ChannelFlowBuilder has to provide to its the block the special implementation of ProducerScope interface that makes send to behave like offer.

Seeing that `ProducerScope` has both `offer` and `send`, changing the semantics of send when `dropWhileBusy` is fused seems alarming. Send could secretly be an offer in the right instance, which seems sketchy  to me. 

Thoughts? If we were to go this route, I would want it to be thoroughly documented on both `produce/broadcast` as well as `dropWhileBusy`

Reply:
Likewise, to ensure first element delivery the overridden `send` (which is really offer) would need to ensure first element delivery. Should we apply those same constraint when a user calls `offer` within the scope?